backend/controllers/searchWithFilter.test.js

```diff
@@ -0,0 +1,396 @@
+          query: {
+            user: 'Andy',
+            status: 'INACTIVE',
+            groupCode: 'MDI',
+            roleCode: 'ACCESS_ROLE_ADMIN',
+            restrictToActiveGroup: 'false',
+          },
+        }
+
+        const render = jest.fn()
+        await search(req, { render, locals: { user: { maintainAccessAdmin: true } } })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.objectContaining({
+            currentFilter: {
+              groupCode: ['MDI'],
+              roleCode: ['ACCESS_ROLE_ADMIN'],
+              status: 'INACTIVE',
+              user: 'Andy',
+              restrictToActiveGroup: false,
+            },
+            downloadUrl:
+              '/search-with-filter-dps-users/download?user=Andy&status=INACTIVE&roleCode=ACCESS_ROLE_ADMIN&groupCode=MDI&activeCaseload=',
+          }),
+        )
+      })
+
+      it('should set list current filters to undefined when no values', async () => {
+        const req = {
+          ...standardReq,
+          query: { user: 'Andy', status: 'INACTIVE', groupCode: '', roleCode: '', restrictToActiveGroup: '' },
+        }
+
+        const render = jest.fn()
+        await search(req, { render, locals: { user: { maintainAccessAdmin: true } } })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.objectContaining({
+            currentFilter: {
+              groupCode: undefined,
+              roleCode: undefined,
+              status: 'INACTIVE',
+              user: 'Andy',
+              restrictToActiveGroup: true,
+            },
+            downloadUrl:
+              '/search-with-filter-dps-users/download?user=Andy&status=INACTIVE&roleCode=&groupCode=&activeCaseload=',
+          }),
+        )
+      })
+
+      it('should set current filter with multiple role and prison query parameters', async () => {
+        const req = {
+          ...standardReq,
+          query: {
+            user: 'Andy',
+            status: 'INACTIVE',
+            groupCode: ['MDI', 'BXI'],
+            roleCode: ['ACCESS_ROLE_ADMIN', 'ACCESS_ROLE_GENERAL'],
+            restrictToActiveGroup: 'true',
+          },
+        }
+
+        const render = jest.fn()
+        await search(req, { render, locals: { user: { maintainAccessAdmin: true } } })
+        expect(render).toBeCalledWith(
+          'searchWithFilter.njk',
+          expect.objectContaining({
+            currentFilter: {
+              groupCode: ['MDI', 'BXI'],
+              roleCode: ['ACCESS_ROLE_ADMIN', 'ACCESS_ROLE_GENERAL'],
+              status: 'INACTIVE',
+              user: 'Andy',
+              restrictToActiveGroup: true,
+            },
+            downloadUrl:
+              '/search-with-filter-dps-users/download?user=Andy&status=INACTIVE&roleCode=ACCESS_ROLE_ADMIN&roleCode=ACCESS_ROLE_GENERAL&groupCode=MDI&groupCode=BXI&activeCaseload=MDI&activeCaseload=BXI',
+          }),
+        )
```

Comment:
demonstrate that multiple roles/prison would be passed even though UI doesn't allow it yet and API wouldn't work anyway, but showing it can be supported